coral-component-shell/src/scripts/ShellMenuBarItem.js

```diff
@@ -214,13 +248,47 @@ class ShellMenuBarItem extends BaseComponent(HTMLElement) {
     }
     
     // Link menu with item
-    if (menu) {
+    if (menu !== null) {
       this.id = this.id || commons.getUID();
       menu.setAttribute('target', `#${this.id}`);
-      const shellMenuButton = this._elements.shellMenuButton;
-      shellMenuButton.setAttribute('aria-haspopup', menu.getAttribute('role') || 'dialog');
+      if (this.hasPopup === hasPopupRole.DEFAULT) {
+        this.hasPopup = menu.getAttribute('role') || 'dialog';
+      }
+    }
+    else if (this._menu && this.hasPopup !== hasPopupRole.DEFAULT) {
+      this.hasPopup = hasPopupRole.DEFAULT;
+    }
+  }
+
+  /**
+    Whether the item opens a popup dialog or menu. Accepts either "menu", "listbox", "tree", "grid", or "dialog".
+    @type {?String}
+    @default ShellMenuBarItemHasPopupRoleEnum.DEFAULT
+    @htmlattribute haspopup
+  */
+  get hasPopup() {
+    return this._hasPopup || null;
+  }
+  set hasPopup(value) {
+    value = transform.string(value).toLowerCase();
+    this._hasPopup = validate.enumeration(hasPopupRole)(value) && value || hasPopupRole.DEFAULT;
+
+    const shellMenuButton = this._elements.shellMenuButton;
+    let ariaHaspopup = this._hasPopup;
+
+    // When hasPopup equals "true", the default aria-haspopup role for a shellMenuButton should be "dialog".
+    if (ariaHaspopup === hasPopupRole.TRUE) {
```

Comment:
Why would someone use TRUE if you can use DIALOG especially since there's no difference between both ?

Reply:
`true` is a backward-compatible value for `aria-haspopup` between WAI-ARIA 1.1 and WAI-ARIA 1.0, but for our purposes, we probably don't need to support it as a value of `hasPopup`.

Reply:
`true` is a backward-compatible value for `aria-haspopup` that usually interpreted by AT as a `menu`. The attribute name `aria-haspopup` sounds like a boolean property, and in WAI-ARIA 1.0, the only acceptable values were `true` and `false`, so mapping `hasPopup="true"` to `aria-haspopup="dialog"` is an attempt to normalize the value.

Reply:
I can remove this.